test/cpp/ext/otel/otel_tracing_test.cc

```
@@ -0,0 +1,545 @@
+                opentelemetry::exporter::memory::InMemorySpanExporterFactory::
+                    Create(data_)));
+    tracer_ = tracer_provider->GetTracer("grpc-test");
+    ASSERT_TRUE(OpenTelemetryPluginBuilder()
+                    .SetTracerProvider(std::move(tracer_provider))
+                    .SetTextMapPropagator(
+                        grpc::experimental::MakeGrpcTraceBinTextMapPropagator())
+                    .BuildAndRegisterGlobal()
+                    .ok());
+    grpc::ServerBuilder builder;
+    int port;
+    // Use IPv4 here because it's less flaky than IPv6 ("[::]:0") on Travis.
+    builder.AddListeningPort("0.0.0.0:0", grpc::InsecureServerCredentials(),
+                             &port);
+    builder.RegisterService(&service_);
+    server_ = builder.BuildAndStart();
+    server_address_ = absl::StrCat("localhost:", port);
+    auto channel = grpc::CreateChannel(server_address_,
+                                       grpc::InsecureChannelCredentials());
+    stub_ = EchoTestService::NewStub(channel);
+  }
+
+  void TearDown() override {
+    server_->Shutdown();
+    grpc_shutdown_blocking();
+    grpc_core::ServerCallTracerFactory::TestOnlyReset();
+    grpc_core::GlobalStatsPluginRegistryTestPeer::
+        ResetGlobalStatsPluginRegistry();
+  }
+
+  void SendRPC() {
+    EchoRequest request;
+    request.set_message("foo");
+    EchoResponse response;
+    grpc::ClientContext context;
+    grpc::Status status = stub_->Echo(&context, request, &response);
+  }
+
+  opentelemetry::nostd::shared_ptr<opentelemetry::trace::Tracer> tracer_;
+  std::shared_ptr<opentelemetry::exporter::memory::InMemorySpanData> data_;
+  CallbackTestServiceImpl service_;
+  std::string server_address_;
+  std::unique_ptr<grpc::Server> server_;
+  std::unique_ptr<EchoTestService::Stub> stub_;
+};
+
+TEST_F(OTelTracingTest, Basic) {
+  SendRPC();
+  absl::SleepFor(absl::Milliseconds(500 * grpc_test_slowdown_factor()));
```

I assume these sleeps are needed to let the data propagate into OT?  How do we know this sleep is long enough?